src/core/Form/TimeInput/TimeInput.tsx

```diff
@@ -0,0 +1,300 @@
+import React, {
+  forwardRef,
+  ChangeEvent,
+  ReactNode,
+  ReactElement,
+  useRef,
+  useState,
+} from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { AutoId } from '../../utils/AutoId/AutoId';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
+import { Debounce } from '../../utils/Debounce/Debounce';
+import { getConditionalAriaProp } from '../../../utils/aria';
+import {
+  HTMLAttributesIncludingDataAttributes,
+  forkRefs,
+} from '../../../utils/common/common';
+import {
+  spacingStyles,
+  separateMarginProps,
+  MarginProps,
+} from '../../theme/utils/spacing';
+import { HtmlInputProps, HtmlDiv, HtmlSpan, HtmlInput } from '../../../reset';
+import { Label, LabelMode } from '../Label/Label';
+import { StatusText } from '../StatusText/StatusText';
+import { HintText } from '../HintText/HintText';
+import { InputStatus, StatusTextCommonProps } from '../types';
+import { baseStyles } from './TimeInput.baseStyles';
+
+const baseClassName = 'fi-time-input';
+export const timeInputClassNames = {
+  baseClassName,
+  fullWidth: `${baseClassName}--full-width`,
+  disabled: `${baseClassName}--disabled`,
+  error: `${baseClassName}--error`,
+  success: `${baseClassName}--success`,
+  labelIsVisible: `${baseClassName}_label--visible`,
+  icon: `${baseClassName}_with-icon`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  inputElement: `${baseClassName}_input`,
+  styleWrapper: `${baseClassName}_wrapper`,
+  statusTextHasContent: `${baseClassName}_statusText--has-content`,
+};
+
+export interface TimeInputProps
+  extends StatusTextCommonProps,
+    MarginProps,
+    Omit<HtmlInputProps, 'type' | 'onChange' | 'onBlur' | 'defaultValue'> {
+  /** CSS class for custom styles */
+  className?: string;
+  /** Props passed to the outermost div element of the component */
+  wrapperProps?: Omit<
+    HTMLAttributesIncludingDataAttributes<HTMLDivElement>,
+    'className'
+  >;
+  /** Disables the input */
+  disabled?: boolean;
+  /** Callback fired on input click */
+  onClick?: () => void;
+  /** Callback fired on input change */
+  onChange?: (value: string) => void;
+  /** Callback fired on input blur.
+   * Notice that this function returns the string value of the input,
+   * not the blur FocusEvent like in other input components in this library!
+   */
+  onBlur?: (inputValue: string) => void;
+  /** Label for the input */
+  labelText: ReactNode;
+  /** Hides or shows the label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: LabelMode;
+  /** Hint text to be shown below the component */
```

Comment:
Should probably say 'between the label and the input' or something of the sort?

Reply:
Yes, fixed!